libraries/AP_Scripting/AP_Scripting_SerialDevice.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+  port for a script to access from a device perspective
+ */
+
+#include "AP_Scripting_config.h"
+#include "AP_Scripting.h"
+
+#if AP_SCRIPTING_ENABLED && AP_SCRIPTING_SERIALDEVICE_ENABLED
+
+#include <AP_Math/AP_Math.h>
+#include <AP_BoardConfig/AP_BoardConfig.h>
+
+#ifndef AP_SCRIPTING_SERIALDEVICE_MIN_TXSIZE
+#define AP_SCRIPTING_SERIALDEVICE_MIN_TXSIZE 2048
+#endif
+
+#ifndef AP_SCRIPTING_SERIALDEVICE_MIN_RXSIZE
+#define AP_SCRIPTING_SERIALDEVICE_MIN_RXSIZE 2048
+#endif
+
+/*
+  initialise scripting serial ports
+*/
+void AP_Scripting_SerialDevice::init(void)
+{
+    if (enable == 0) {
+        return;
+    }
+
+    for (uint8_t i=0; i<ARRAY_SIZE(ports); i++) {
+        auto &p = ports[i];
```

Comment:
```suggestion
    for (auto &p : ports) {
```

similarly elsewhere

Reply:
I changed elsewhere, but in this particular case I need `i` in the loop.